src/lib/mdns/minimal/records/ResourceRecord.h

```diff
@@ -51,15 +54,12 @@ class ResourceRecord
     /// Output the data portion of the resource record.
     virtual bool WriteData(chip::BufBound & out) const = 0;
 
-    ResourceRecord(QType type, const QNamePart * names, uint16_t namesCount) : mType(type), mQNameCount(namesCount), mQName(names)
-    {}
+    ResourceRecord(QType type, FullQName name) : mType(type), mQName(name) {}
 
 private:
     const QType mType;
-    uint64_t mTtl = 0;
-
-    const uint16_t mQNameCount;
-    const QNamePart * mQName;
+    uint64_t mTtl = kDefaultTtl;
```

Comment:
Need a TODO for TTL handling. TTL handling will be *critical* in CHIP, but the implementation conversations have not happened. Overall, we need to make TTL configurable, so that anything sleepy or low-power can set a really long TTL, and most records are long-ish TTL to avoid constant re-discovery. Most nodes are expected to be available steady state and failure to communicate would cause eviction of cache entry and re-discovery, which would handle dealing with TTLs that are quite long.

Reply:
I would dever this for later.
Server side can for sure change TTL (even 0 may be useful to clear it) but current code does not have these concerns. The server is trying to let the app control the TTL.